services/zone-service.ts

```diff
@@ -73,30 +73,35 @@ class ZoneService {
             sterilizedCats: []
         });
         
-        for(const unsterilizedCat of interestZone.unsterilizedCats) {
-            await this.validateUnsterilizedCat(unsterilizedCat);
-            const catSchema = new unsterilizedCatSchema(unsterilizedCat);
-            newZoneSchema.unsterilizedCats.push(catSchema);
+        if (interestZone.unsterilizedCats) {
+            for(const unsterilizedCat of interestZone.unsterilizedCats) {
+                await this.validateUnsterilizedCat(unsterilizedCat);
+                const catSchema = new unsterilizedCatSchema(unsterilizedCat);
+                newZoneSchema.unsterilizedCats.push(catSchema);
+            }
         }
 
-        for(const sterilizedCat of interestZone.sterilizedCats) {
-            await this.validateSterilizedCat(sterilizedCat);
-            const catSchema = new sterilizedCatSchema(sterilizedCat);
-            newZoneSchema.sterilizedCats.push(catSchema);
+        if (interestZone.sterilizedCats) {
+            for(const sterilizedCat of interestZone.sterilizedCats) {
+                await this.validateSterilizedCat(sterilizedCat);
+                const catSchema = new sterilizedCatSchema(sterilizedCat);
+                newZoneSchema.sterilizedCats.push(catSchema);
+            }
         }
 
         return await newZoneSchema.save();        
     }
 
-    async addCatToZone(zoneID: string | string[], cat: any) {
-        const interestZone = await this.findById(zoneID);
+    async addCatToZone(zoneID: string | string[], cat: any, interestZone?: any) {
+        if (interestZone === undefined) interestZone = await this.findById(zoneID);
         const sterilizedStatus: boolean = await this.validateCat(cat);
 
         if (sterilizedStatus) {
             const catSchema = new sterilizedCatSchema(cat);
             interestZone.sterilizedCats.push(catSchema);
         } else {
             const catSchema = new unsterilizedCatSchema(cat);
+            interestZone.noUnsterilizedCats++;
```

Comment:
Can you please confirm that the field above is necessary? At first glance, it can be deduced from `interestZone.unsterilizedCats.length`